templates/external_auth_api.conf.erb

```diff
@@ -0,0 +1,28 @@
+
+<LocationMatch ^/api(/v2)?/users/extlogin/?$>
+  SSLRequireSSL
+  <% if scope.lookupvar('foreman::ipa_authentication_api') %>
+  <If "%{HTTP:Authorization} =~ /^Basic/">
+    AuthType Basic
+    AuthName "PAM Authentication"
+    AuthBasicProvider PAM
```

Comment:
I think this implies the pam module should be included. Note that https://httpd.apache.org/docs/2.4/mod/mod_auth_basic.html#authbasicprovider doesn't list it as a built in provider, but https://www.adelton.com/apache/mod_authnz_pam/ does document what you're doing here.

There is `apache::mod::authnz_pam` which will ensure the correct package is installed and the module is loaded.